lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -5635,7 +5635,9 @@ void LLVMToSPIRVBase::transFunction(Function *I) {
 
   if (isKernel(I)) {
     auto Interface = collectEntryPointInterfaces(BF, I);
-    BM->addEntryPoint(ExecutionModelKernel, BF->getId(), BF->getName(),
+    auto Name =
+        BM->getGenerateKernelEntryPoints() ? BF->getName() : I->getName().str();
+    BM->addEntryPoint(ExecutionModelKernel, BF->getId(), Name,
```

Comment:
Is this change necessary?  If !getGenerateKernelEntryPoints, then BF->getName()==I->getName().str()  since BF->getName() will be set by line 885.